src/gui/qgsexpressionselectiondialog.cpp

```diff
@@ -103,6 +118,63 @@ void QgsExpressionSelectionDialog::on_mActionRemoveFromSelection_triggered()
   saveRecent();
 }
 
+void QgsExpressionSelectionDialog::on_mButtonZoomToSelection_clicked()
+{
+  if ( mExpressionBuilder->expressionText().isEmpty() )
+    return;
+
+  QgsFeatureIds ids;
+
+  QgsExpressionContext context;
+  context << QgsExpressionContextUtils::globalScope()
+  << QgsExpressionContextUtils::projectScope()
+  << QgsExpressionContextUtils::layerScope( mLayer );
+
+  QgsFeatureRequest request = QgsFeatureRequest().setFilterExpression( mExpressionBuilder->expressionText() )
+                              .setExpressionContext( context )
+                              .setSubsetOfAttributes( QgsAttributeList() );
+
+  QgsFeatureIterator features = mLayer->getFeatures( request );
+
+  QgsRectangle bbox;
+  bbox.setMinimal();
+  QgsFeature feat;
+  int featureCount = 0;
+  while ( features.nextFeature( feat ) )
+  {
+    QgsGeometry geom = feat.geometry();
+    if ( geom.isEmpty() || geom.geometry()->isEmpty() )
+      continue;
+
+    QgsRectangle r = mMapCanvas->mapSettings().layerExtentToOutputExtent( mLayer, geom.boundingBox() );
+    bbox.combineExtentWith( r );
+    featureCount++;
+  }
+  features.close();
+
+  QSettings settings;
+  int timeout = settings.value( QStringLiteral( "/qgis/messageTimeout" ), 5 ).toInt();
+  if ( featureCount > 0 )
+  {
+    mMapCanvas->zoomToFeatureExtent( bbox );
```

Comment:
Good point; I'll do a null check, and hide the "zoom to features" button if no map canvas is set.